lib/hydra/derivatives/processors/active_encode.rb

```diff
@@ -0,0 +1,53 @@
+require 'active_encode'
+
+module Hydra::Derivatives::Processors
+  class ActiveEncodeError < StandardError
+    def initialize(status, source_path, errors = [])
+      msg = "ActiveEncode status was \"#{status}\" for #{source_path}"
+      msg = "#{msg}: #{errors.join(' ; ')}" unless errors.empty?
+      super(msg)
+    end
+  end
+
+  class ActiveEncode < Processor
+    class_attribute :timeout
+    attr_accessor :encode_class
+
+    def initialize(source_path, directives, opts = {})
+      super
+      @encode_class = opts.delete(:encode_class) || ::ActiveEncode::Base
+    end
+
+    def process
+      encode = encode_class.create(source_path, directives)
+      timeout ? wait_for_encode_with_timeout(encode) : wait_for_encode(encode)
+      encode.output.each do |output|
+        output_file_service.call(output, directives)
+      end
+    end
+
+    def wait_for_encode_with_timeout(encode)
```

Comment:
Should this method be part of the public interface or should it  and the methods below be private?